### PR TITLE
remove `Deployment.is_schedule_active` and `Deployment.schedule`

### PR DESCRIPTION
**Pull request by jakekaplan**
closes: https://github.com/PrefectHQ/prefect/issues/14110

This PR removes `Deployment.is_schedule_active` and `Deployment.schedule` as they are past their deprecation window in favor of `Deployment.paused` and `Deployment.schedules`. This includes

- Removing `is_schedule_active` and `schedule` fields from both client and server side `Deployment` models
- Removing `is_schedule_active` and `schedule` from the `RunnerDeployment`s
- Updating `client.schemas.actions.DeploymentUpdate` to use `paused`/`scheduled`
- Removing various compatibility functions in endpoints and models
- Removing `DeploymentFilterIsScheduleActive` in favor of `DeploymentFilterPaused`
- Removing `/.../set_schedule_active` in favor of  `/.../resume_deployment`
- Removing `/.../set_schedule_inactive` in favor of `/.../pause_deployment`
- Updating `PrefectClient.update_deployment` to match our other existing update methods
- Fixing/removing tests to utilize the `paused`/`schedules`

### CODE REVIEW

File: src/prefect/server/database/orm_models.py [hunk 1]
```diff
@@ -865,11 +865,6 @@ def job_variables(self):
         nullable=True,
         index=True,
     )
-
-    schedule = sa.Column(Pydantic(schemas.schedules.SCHEDULE_TYPES))
```
Comment (jakekaplan) [inline src/prefect/server/database/orm_models.py:869]: We're dropping these from the ORM layer. In theory we've already migrated everyone to use `schedules` and `paused` back with the migration that added those. Should we drop these in a migration?

File: src/prefect/client/orchestration.py [hunk 2]
```diff
@@ -1691,53 +1682,19 @@ async def create_deployment(
...
         await self._client.patch(
             f"/deployments/{deployment_id}", json={"paused": paused}
         )
 
     async def update_deployment(
         self,
-        deployment: Deployment,
-        schedule: SCHEDULE_TYPES = None,
-        is_schedule_active: Optional[bool] = None,
+        deployment_id: UUID,
```
Comment (jakekaplan) [inline src/prefect/client/orchestration.py:1692]: This is more consistent with our other update methods so you can actually do a partial update

**Comment by jakekaplan** [comment]
@cicdw could you approve this when you get a sec?

**Review by cicdw — APPROVED** [review]

**Review by discdiver — APPROVED** [review]
LGTM. Separately, is it worth mentioning this update in the 2->3 upgrade doc?

**Pull request merged**